robot_modeling/DQ_SerialManipulatorDH.m

```diff
@@ -63,76 +64,8 @@
         % Prismatic joint
         JOINT_PRISMATIC = 2;
     end
-    
-    methods
-        function obj = DQ_SerialManipulatorDH(A,convention)
-            % These are initialized in the constructor of
-            % DQ_SerialManipulator
-            %obj.convention = convention;
-            %obj.n_links = size(A,2);
-            %obj.theta = A(1,:);
-            %obj.d = A(2,:);
-            %obj.a = A(3,:);
-            %obj.alpha = A(4,:);
-            obj = obj@DQ_SerialManipulator(A(1:4,:),convention);
-            
-            if nargin == 0
-                error('Input: matrix whose columns contain the DH parameters')
-            end
-            
-            if(size(A,1) ~= 5)
-                error('Input: Invalid DH matrix. It should have 5 rows.')
-            end
-            
-            % Add type
-            obj.type = A(5,:);
-        end
-        
-        function x = raw_fkm(obj,q,to_ith_link)
-            %   RAW_FKM(q) calculates the forward kinematic model and
-            %   returns the dual quaternion corresponding to the
-            %   last joint (the displacements due to the base and the effector
-            %   are not taken into account).
-            %
-            %   'q' is the vector of joint variables
-            %   'to_ith_link' defines until which link the raw_fkm will be
-            %   calculated.
-            %
-            %   This is an auxiliary function to be used mainly with the
-            %   Jacobian function.
-            if nargin == 3
-                n = to_ith_link;
-            else
-                n = obj.n_links;
-            end
-            
-            x = DQ(1);
-            
-            for i=1:n
-                x = x*dh2dq(obj,q(i),i);
-            end
-        end
-        
-        function x = fkm(obj,q,to_ith_link)
-            %   FKM(q) calculates the forward kinematic model and
-            %   returns the dual quaternion corresponding to the
-            %   end-effector pose. This function takes into account the
-            %   displacement due to the base's and effector's poses.
-            %
-            %   'q' is the vector of joint variables
-            %   'to_ith_link' defines up to which link the fkm will be
-            %   calculated. If to_ith_link corresponds to the last link,
-            %   the method DOES NOT take into account the transformation
-            %   given by set_effector. If you want to take into account
-            %   that transformation, use FKM(q) instead.
-            
-            if nargin == 3
-                x = obj.reference_frame*obj.raw_fkm(q, to_ith_link); %Takes into account the base displacement
-            else
-                x = obj.reference_frame*obj.raw_fkm(q)*obj.effector;
-            end
-        end
-        
+
+    methods (Access = protected)
         function dq = dh2dq(obj,q,ith)
             %   For a given link's Extended DH parameters, calculate the correspondent dual
```

Comment:
What is an "extended" DH parameters? 

correspondent-> corresponding

Also, please include the bibliographic reference to this method as I instructed in my comments to the abstract method.

Reply:
> What is an "extended" DH parameters?
> 
> correspondent-> corresponding
> 
> Also, please include the bibliographic reference to this method as I instructed in my comments to the abstract method.

I do not know, I did not add that comment. I'm going to improve the documentation of that method.

Reply:
- [ ] Implemented.